### PR TITLE
Editorial: fixed some minor issues and move the rest spec to modern …

### PR DESCRIPTION
**Pull request by Honry**
…ReSpec

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -176,7 +176,7 @@
   <p>
     The term <dfn>obtain permission</dfn> for a certain operation indicates
     that the UA has either obtained <a>expressed permission</a>, or
-    <a href="#askforgiveness">asked for forgiveness</a>, or ensured a
+    <a href="#dfn-ask-for-forgiveness">asked for forgiveness</a>, or ensured a
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
    <a>asked for forgiveness</a>, or ensured a
```

File: index.html [hunk 2]
```diff
@@ -1395,7 +1395,7 @@ <h4>
     <h2>The <dfn>NDEFRecordType</dfn> enum</h2>
       <p>
         This enum defines the set of known types for a <a>NDEFRecord</a>. The
-        <a href="#data-mapping">data mapping</a> section describes how
+        <a href="#data-mapping"></a> section describes how
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
        [[[#data-mapping]]] section describes how
```

File: index.html [hunk 3]
```diff
@@ -1486,15 +1486,15 @@ <h2>The <dfn>NDEFRecordType</dfn> enum</h2>
       <td><i>empty</i></td>
       <td>
         {{ArrayBuffer}} or<br>
-        any <a href="!WEBIDL#dfn-typed-array-type"></a>
+        any <a data-cite="webidl">typed array type</a>
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
        any <a>typed array type</a>
```

File: index.html [hunk 4]
```diff
@@ -1486,15 +1486,15 @@ <h2>The <dfn>NDEFRecordType</dfn> enum</h2>
...
       </td>
       <td><a>External type record</a></td>
     </tr>
   </table>
   <p>
     The mapping from <a>NDEF record</a> types to <a>NDEFRecord</a>,
     as used for incoming <a>NDEF message</a>s described in the
-    <a href="#steps-receiving">Receiving and parsing content</a> section, is as
+    <a href="#steps-receiving"></a> section, is as
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
    [[[#steps-receiving]]] section, is as
```

File: index.html [hunk 5]
```diff
@@ -2467,154 +2467,153 @@ <h3>The <dfn>NFCReaderOptions</dfn> dictionary</h3>
         </p>
         <ol>
           <li>
-            If the type of a <var>record</var>'s data is not a {{DOMString}}
-            or an unrestricted {{double}}, [= exception/throw =] a {{"TypeError"}}
+            If the type of a |record|'s data is not a {{DOMString}}
+            or an {{unrestricted double}}, [= exception/throw =] a {{"TypeError"}}
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
            or an {{unrestricted double}}, [= exception/throw =] a {{TypeError}}
```

File: index.html (hunk 1 above)
Comment (Honry) [reply index.html]: The `dfn` is "ask for forgiveness",  use `<a>asked for forgiveness</a>` will throw ReSpec error.

File: index.html (hunk 5 above)
Comment (marcoscaceres) [reply index.html]: Note, only DOMExceptions are named... TypeError and other ES errors are not :)

File: index.html [hunk 6]
```diff
@@ -2467,154 +2467,153 @@ <h3>The <dfn>NFCReaderOptions</dfn> dictionary</h3>
...
 
       <section><h3>Mapping URL to NDEF</h3>
       <p>
-        To <dfn>map a URL to NDEF</dfn> given a <var>record</var>, run these
+        To <dfn>map a URL to NDEF</dfn> given a |record:NDEFRecordInit|, run these
         steps:
         <ol>
           <li>
-            If <var>record</var>'s data is not a <a>string</a>,
+            If |record|'s data is not a <a>string</a>,
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
            If |record|'s data is not a {{DOMString}},
```

File: index.html (hunk 2 above)
Comment (Honry) [reply index.html]: No, this will not work.

File: index.html (hunk 3 above)
Comment (Honry) [reply index.html]: Will fixed. :)

File: index.html (hunk 4 above)
Comment (Honry) [reply index.html]: As above, this will not work.

File: index.html [hunk 7]
```diff
@@ -2467,154 +2467,153 @@ <h3>The <dfn>NFCReaderOptions</dfn> dictionary</h3>
...
             [= exception/throw =] a {{"TypeError"}} and abort these steps.
           </li>
           <li>
-            Let <var>urlRecord</var> be the result of
-            <a data-lt="url parser">parsing</a> <var>record</var>'s data.
+            Let |urlRecord:string| be the result of
```
Comment (marcoscaceres) [inline index.html]: Check what the parser actually returns... it's probably a USVString, but check! 
```suggestion
            Let |urlRecord:USVString| be the result of
```

Comment (marcoscaceres) [reply index.html]: Also, this is weird... why is called a "record" if it's a URL?

File: index.html (hunk 5 above)
Comment (Honry) [reply index.html]: Thanks, got it!

File: index.html [hunk 8]
```diff
@@ -2467,154 +2467,153 @@ <h3>The <dfn>NFCReaderOptions</dfn> dictionary</h3>
...
+            <a data-lt="url parser">parsing</a> |record|'s data.
           </li>
           <li>
-            If <var>urlRecord</var> is failure, [= exception/throw =] a
+            If |urlRecord| is failure, [= exception/throw =] a
```
Comment (marcoscaceres) [inline index.html:2578]: SyntaxError seems like an odd choice to me...

File: index.html [hunk 9]
```diff
@@ -2927,8 +2926,7 @@ <h3>The <dfn>NFCReaderOptions</dfn> dictionary</h3>
 
     <section> <h3>The <strong>start()</strong> method</h3>
       <p>
-        The section <a href="#steps-receiving">
-        Receiving and parsing content</a> uses <a>NFCReader</a> instances to match
+        The section <a href="#steps-receiving"></a> uses <a>NFCReader</a> instances to match
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
        The section [[[#steps-receiving]]] uses {{NFCReader}} instances to match
```

File: index.html [hunk 10]
```diff
@@ -3228,117 +3223,118 @@ <h3>Receiving and parsing content</h3>
...
               <li>
-                Let <var>buffer</var> be the <a>byte sequence</a> of <var>
-                ndef</var>'s <a>PAYLOAD field</a> from the second <a>byte</a> to the end.
+                Let |buffer:bytes| be the <a>byte sequence</a> of 
+                |ndef|'s <a>PAYLOAD field</a> from the second <a>byte</a> to the end.
               </li>
               <li>
-                <a data-cite="encoding#concept-stream-push">Push</a> <var>buffer</var>
-                to <var>record</var>'s associated <a>bytes</a>.
+                <a data-cite="encoding#concept-stream-push">Push</a> |buffer|
```
Comment (marcoscaceres) [inline index.html:3235]: 🚨 you need to figure out why Encoding didn't export this.

File: index.html (hunk 1 above)
Comment (marcoscaceres) [reply index.html]: ah, change the dfn to:
```
<dfn data-lt="asked for forgiveness">ask for forgiveness</dfn> 
```

Then you can use the alias "asked for forgiveness" :)

Comment (Honry) [reply index.html]: Thanks, that's what I want.

File: index.html (hunk 6 above)
Comment (Honry) [reply index.html]: Thanks, will fix it.

File: index.html (hunk 7 above)
Comment (Honry) [reply index.html]: URL is one of the NDEF record, I suppose that's why it was named as "urlRecord".

File: index.html (hunk 8 above)
Comment (Honry) [reply index.html:2578]: Do you have better choice?

File: index.html (hunk 10 above)
Comment (Honry) [reply index.html:3235]: Files https://github.com/whatwg/encoding/issues/180.

File: index.html (hunk 2 above)
Comment (marcoscaceres) [reply index.html]: Odd... what's it say?

File: index.html (hunk 8 above)
Comment (marcoscaceres) [reply index.html:2578]: I think it's just a TypeError.

Comment (marcoscaceres) [reply index.html:2578]: (See also what the URL spec does... it seems to also throw TypeError)

Comment (Honry) [reply index.html:2578]: OK, I will file a separated issue.

File: index.html (hunk 2 above)
Comment (Honry) [reply index.html]: I may find the root cause, it requires an `id` definition to the corresponding section.

Comment (marcoscaceres) [reply index.html]: Ah, yeah :) Good point. Forgot about that.

**Review by marcoscaceres — APPROVED** [review]
Looking pretty good! 👍

File: index.html [hunk 11]
```diff
@@ -2467,154 +2467,153 @@ <h3>The <dfn>NFCReaderOptions</dfn> dictionary</h3>
...
           </li>
           <li>
             Match the URI prefixes as defined in [[[NFC-STANDARDS]]]
             URI Record Type Definition specification, Section 3.2.2, against
-            the <var>urlRecord</var>.
+            the |urlRecord|.
           </li>
           <li>
-            Let <var>prefix</var> be the matched prefix or else the
+            Let |prefix:string| be the matched prefix or else the
```
Comment (kenchris) [inline index.html:2598]: Is string correct? Or should we use USVString @marcoscaceres

**Review by kenchris — APPROVED** [review]

File: index.html (hunk 11 above)
Comment (marcoscaceres) [reply index.html:2598]: I'm not sure what prefix is in this context... if it's part of a URL, then a USVString might be suitable... if it's just a random string, then string is fine.

**Pull request merged**